### ISSUE TITLE
about _sigmoid()

### ISSUE THREAD

**Issue by hktxt** [xingyizhou/CenterNet#201]
Can some one explain to me why apply _sigmoid() to optput['hm'] before computing loss
```
def _sigmoid(x):
  y = torch.clamp(x.sigmoid_(), min=1e-4, max=1-1e-4)
  return y
```
`output['hm'] = _sigmoid(output['hm'])`

https://github.com/xingyizhou/CenterNet/blob/master/src/lib/trains/ctdet.py#L34

**Comment by xingyizhou** [comment]
This is defined in our loss. I haven't tried removing it.

**Comment by joyyang1215** [comment]
If you use mse_loss, you do not need to apply sigmoid.
But, if you use focal_loss defined in CenterNet's paper, the heatmap output must within 0 - 1; otherwise, it does not work.

**Comment by wuzuowuyou** [comment]
why clamp(x.sigmoid_(), min=1e-4, max=1-1e-4)?
1e-4
